homeassistant/components/sensor/netatmo.py

```diff
@@ -43,29 +36,9 @@
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Setup the NetAtmo sensor."""
-    if not validate_config({DOMAIN: config},
-                           {DOMAIN: [CONF_API_KEY,
-                                     CONF_USERNAME,
-                                     CONF_PASSWORD,
-                                     CONF_SECRET_KEY]},
-                           _LOGGER):
-        return None
-
-    import lnetatmo
-
-    authorization = lnetatmo.ClientAuth(config.get(CONF_API_KEY, None),
-                                        config.get(CONF_SECRET_KEY, None),
-                                        config.get(CONF_USERNAME, None),
-                                        config.get(CONF_PASSWORD, None))
-
-    if not authorization:
-        _LOGGER.error(
-            "Connection error "
-            "Please check your settings for NatAtmo API.")
-        return False
-
-    data = NetAtmoData(authorization, config.get(CONF_STATION, None))
+    """Setup the available NetAtmo weather sensors."""
+    netatmo = get_component('netatmo')
+    data = NetAtmoData(netatmo.NETATMO_AUTH, config.get(CONF_STATION, None))
```

Comment:
This parameter was newly added to the Netatmo sensor it allows to limit the sensors to a specific Weather Station, but if None is provided then all user's weather stations will be parsed. I did not work on the PR that was introducing this change, but I confirm that it did not break a thing